Content.Server/GameTicking/Rules/RevolutionaryRuleSystem.cs

```diff
@@ -0,0 +1,392 @@
+    /// </summary>
+    public void OnPostFlash(EntityUid uid, HeadRevolutionaryComponent comp, ref AfterFlashedEvent ev)
+    {
+        var stunTime = TimeSpan.FromSeconds(3);
+        if (!HasComp<RevolutionaryComponent>(ev.Target) && !HasComp<MindShieldComponent>(ev.Target) && (HasComp<HumanoidAppearanceComponent>(ev.Target) || HasComp<MonkeyAccentComponent>(ev.Target))
+            && TryComp<MobStateComponent>(ev.Target, out var mobState) && mobState.CurrentState == MobState.Alive && !HasComp<ZombieComponent>(ev.Target))
+        {
+            _npcFaction.AddFaction(ev.Target, "Revolutionary");
+            EnsureComp<RevolutionaryComponent>(ev.Target);
+            _sharedStun.TryParalyze(ev.Target, stunTime, true);
+            if (ev.User != null)
+            {
+                _adminLogManager.Add(LogType.Mind, LogImpact.Medium, $"{ToPrettyString(ev.User.Value)} converted {ToPrettyString(ev.Target)} into a Revolutionary");
+            }
+            if (_mindSystem.TryGetMind(ev.Target, out var mindId, out var mind))
+            {
+                if (!_roleSystem.MindHasRole<RevolutionaryRoleComponent>(mindId))
+                {
+                    _roleSystem.MindAddRole(mindId, new RevolutionaryRoleComponent { PrototypeId = "Rev" });
+                }
+                if (mind.Session != null)
+                {
+                    var message = Loc.GetString("rev-role-greeting");
+                    var wrappedMessage = Loc.GetString("chat-manager-server-wrap-message", ("message", message));
+                    _chatManager.ChatMessageToOne(Shared.Chat.ChatChannel.Server, message, wrappedMessage, default, false, mind.Session.ConnectedClient, Color.Red);
+                }
+            }
+        }
+    }
+
+    public void OnHeadRevAdmin(EntityUid mindId, MindComponent mind)
+    {
+        if (mind.OwnedEntity != null)
+        {
+            var player = new List<EntityUid>();
+            player.Add((EntityUid) mind.OwnedEntity);
+            InitialAssignCommandStaff();
+            GiveHeadRev(player, "Rev", null);
+        }
+        if (mind.Session != null)
+        {
+            var message = Loc.GetString("head-rev-role-greeting");
+            var wrappedMessage = Loc.GetString("chat-manager-server-wrap-message", ("message", message));
+            _chatManager.ChatMessageToOne(Shared.Chat.ChatChannel.Server, message, wrappedMessage, default, false, mind.Session.ConnectedClient, Color.FromHex("#5e9cff"));
+        }
+    }
+    private void OnCommandMobStateChanged(EntityUid uid, CommandStaffComponent comp, MobStateChangedEvent ev)
+    {
+        if (ev.NewMobState == MobState.Dead || ev.NewMobState == MobState.Invalid)
+            CheckCommandLose();
+    }
+    /// <summary>
+    /// Checks if all of command is dead and if so will remove all sec and command jobs if there were any left.
+    /// </summary>
+    private bool CheckCommandLose()
+    {
```

Comment:
```suggestion
    }

    private void OnCommandMobStateChanged(EntityUid uid, CommandStaffComponent comp, MobStateChangedEvent ev)
    {
        if (ev.NewMobState == MobState.Dead || ev.NewMobState == MobState.Invalid)
            CheckCommandLose();
    }

    /// <summary>
    /// Checks if all of command is dead and if so will remove all sec and command jobs if there were any left.
    /// </summary>
    private bool CheckCommandLose()
    {
```

for readability, try and do this elsewhere as well, not going to comment on every case of it